build/scripts/DevEnv/NewDevEnv.ps1

```diff
@@ -91,20 +79,18 @@ if($createContainerJob) {
     }
 }
 
-if(Test-ContainerExists -containerName $containerName) {
-    Write-Host "Container $containerName is available" -ForegroundColor Green
-} else {
-    throw "Container $containerName not available. Check if the container was created successfully and is running."
-}
-
-
 # Publish apps
-Write-Host "Publishing apps..." -ForegroundColor Yellow
-$publishingAppsStats = Measure-Command {
-    foreach($currentAppFile in $appFiles) {
-        Publish-BcContainerApp -containerName $containerName -appFile $currentAppFile -syncMode ForceSync -sync -credential $credential -skipVerification -install -useDevEndpoint -replacePackageId
-    }
+if ($appFiles) {
+    Write-Host "Publishing apps..." -ForegroundColor Yellow
+    Publish-Apps -ContainerName $ContainerName -AppFiles $appFiles -Credential $credential
 }
 
-Write-Host "Publishing apps took $($publishingAppsStats.TotalSeconds) seconds"
+# Set up vscode for development against the container (i.e. set up launch.json and settings.json)
+if (-not $SkipVsCodeSetup)
+{
+    Write-Host "Configuring vscode for development against the container..."
+    Configure-ALProjectsInPath -ContainerName $ContainerName -Authentication $Authentication
+}
 
+Write-Host "Development environment setup completed successfully" -ForegroundColor Green
```

Comment:
If the apps failed publishing, then was the dev env setup really successful?

Consider altering the message in case of partial failure. Interesting idea could be to report 3 different statuses: on container creation, on building apps and on publishing apps.

E.g.:
```
✅Container BCApps-123 created successfully
✅Apps app1 and app2 built successfully
❌Failed to publish the following apps:...
```
or
```
❌Container BCApps-123 failed to be created.
✅Apps app1 and app2 built successfully
🟡Skipped publishing apps as container was not created
```

Reply:
Made something like that (ish) 😄 Not as nice but it should still give an accurate representation of what the script did